PLNPlugin.inc.php

```diff
@@ -111,6 +111,31 @@ public function setupComponentHandlers($hookName, $params) {
 		return false;
 	}
 
+	/**
+	 * When the request is supposed to be handled by the plugin, this method will disable:
+	 * - Redirecting non-logged users (the staging server) at contexts protected by login
+	 * - Redirecting non-logged users (the staging server) at non-public contexts to the login page (see more at: PKPPageRouter::route())
+	 */
+	private function _disableRestrictions() {
+		$request = $this->getRequest();
+		// Avoid issues with the APIRouter
+		if (!($request->getRouter() instanceof PageRouter)) {
+			return;
+		}
+
+		$page = $request->getRequestedPage();
+		$operation = $request->getRequestedOp();
+		$arguments = $request->getRequestedArgs();
+		if ([$page, $operation] === ['pln', 'deposits'] || [$page, $operation, $arguments[0] ?? ''] === ['gateway', 'plugin', 'PLNGatewayPlugin']) {
+			define('SESSION_DISABLE_INIT', true);
```

Comment:
I've added this to skip a redirection here: https://github.com/pkp/pkp-lib/blob/main/classes/core/PKPPageRouter.inc.php#L229-L237 (I can add a hook to make it less mysterious if you think it's better).

Anyway, it makes sense to avoid creating sessions for all the plugin requests. But AFAICS plugins are too late in the party to cancel the session (I can take a look if it's tough to enable this possibility).